File: Java/Array/Array Duplicates.java
//https://www.geeksforgeeks.org/problems/find-duplicates-in-an-array/1
class Solution {
    public List<Integer> findDuplicates(int[] arr) {
        // code here
        Map<Integer, Integer> freqMap = new HashMap<>();
        List<Integer> res = new ArrayList<>();

        // Count frequency of each element
        for (int num : arr) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }

        // Add elements that appear more than once
        for (Map.Entry<Integer, Integer> entry : freqMap.entrySet()) {
            if (entry.getValue() > 1) {
                res.add(entry.getKey());
            }
        }

        return res;
    }
}

File: Python/Array/Array Duplicates.py
//https://www.geeksforgeeks.org/problems/find-duplicates-in-an-array/1
class Solution:
    def findDuplicates(self, arr):
        # code here
        res = []
        for i in range(len(arr)):
            count = 1
            for j in range(len(arr)):
                if arr[i] == arr[j] and j != i:
                    count += 1
            if count > 1:
                if arr[i] not in res:
                    res.append(arr[i])
        return res
